### DISCUSSION TITLE
OTP Required in few Clients but not all

### DISCUSSION THREAD

**Discussion by krystadamianpzpn** [keycloak/keycloak#32088] [Q&A]
Hey,
my company want to use OTP, but we came across a problem. So we have ONE REALM with multiple clients, and in few of them we change "browser flow" to custom one with OTP Form Required.
And everything work perfect till....
When some user login in one application where is not required OTP, and next go to service witch required OTP, user are automatically logged (using cookies) without asking about OTP code or whatever. 

And our question is how with multiple clients with stored cookie in browser force keycloak to ask user about OTP code when application need that (client have custom flow)? 
All our application is SPA created with angular, integration with keycloak are created by package "keycloak-js".

**Comment by darius-m** [comment]
I think this can be solved using [step-up authentication](https://www.keycloak.org/docs/latest/server_admin/index.html#_step-up-flow), where you can set up a lower authentication level for some applications (and thus require only password authentication), while others will request higher levels of authentication (with OTP).
